### PR TITLE
refactor(xPlayer): delete getMeta error message

### PR DESCRIPTION
**Pull request by CENSOR1337**
getMeta no longer prints an error message on the index did not exist condition.

### CODE REVIEW

**Comment by CsokiHUN** [comment]
why?

**Comment by NikartikTheBear** [comment]
I believe he meant why would you remove that

**Comment by Gellipapa** [comment]
@CENSOR1337 Hi! Where use hasMeta? 

So not being used anywhere but just an option no one will use it but rely on getMeta() ~= nil or similar checks so I don't think it makes sense to implement it. 

Instead, you could have something that if the server is in debug mode then print returns getMeta with nil of course but then you have debug print which is useful for development.

**Comment by CENSOR1337** [comment]
> @CENSOR1337 Hi! Thank you, I think we can refactor this code to be fully recursive, so the subIndex part could be replaced by a recursive function call and the code wouldn't be so complex. I know you didn't do this, but you are currently modifying this part of the code.

I have made a change base on your suggestion and maintenance a little bit of code flow, to make it easier to read.
Let me know what you think, Thanks.

File: [core]/es_extended/server/classes/player.lua [hunk 1]
```diff
@@ -579,49 +579,43 @@ function CreateExtendedPlayer(playerId, identifier, group, accounts, inventory,
...
-			if type(index) ~= "string" then
-				return print("[^1ERROR^7] xPlayer.getMeta ^5index^7 should be ^5string^7!")
-			end
+        if type(index) ~= "string" then
+            return print("[^1ERROR^7] xPlayer.getMeta ^5index^7 should be ^5string^7!")
+        end
 
-			if self.metadata[index] then
+        local metadata = self.metadata[index]
+        if (metadata == nil) then
```
Comment (Gellipapa) [inline [core]/es_extended/server/classes/player.lua:590]: if not metadata then better.

File: [core]/es_extended/server/classes/player.lua [hunk 2]
```diff
@@ -579,49 +579,43 @@ function CreateExtendedPlayer(playerId, identifier, group, accounts, inventory,
...
-							end
-						end
+            if (_type == "table") then
+                local returnValues = {}
 
-						return returnValues
-					end
+                for i = 1, #subIndex do
+                    local key = subIndex[i]
+                    returnValues[key] = self.getMeta(index, key)
```
Comment (Gellipapa) [inline [core]/es_extended/server/classes/player.lua:607]: key is table break code. You should handle only string.

**Review by Gellipapa — CHANGES_REQUESTED** [review]
Nice so clean better than before version.

**Review by Gellipapa — APPROVED** [review]
Good job, thank you for help.

**Pull request merged**